sites/public/lib/appAutofill.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Address,
+  Application,
+  ApplicationStatus,
+  ApplicationSubmissionType,
+  Language,
+} from "@bloom-housing/backend-core/types"
+
+class AutofillCleaner {
+  application: Application = null
+
+  constructor(application: Application) {
+    this.application = application
+  }
+
+  clean() {
+    // prettier-ignore
+    this.
+      addDefaults().
+      removeAdditionalKeys().
+      removeLiveWorkAddresses()
+
+    return this.application
+  }
+
+  addDefaults() {
+    ;["id", "createdAt", "updatedAt", "deletedAt", "listing", "submissionDate"].forEach((key) => {
+      delete this.application[key]
+    })
+
+    this.application["confirmationId"] = "" // only used on frontend
+    this.application["completedSections"] = 0 // only used on frontend
+    this.application["autofilled"] = true // only used on frontend
+    this.application.submissionType = ApplicationSubmissionType.electronical
+    this.application.language = Language.en
+    this.application.acceptedTerms = false
+    this.application.status = ApplicationStatus.submitted
+    this.application.preferences = []
+
+    return this
+  }
+
+  removeAdditionalKeys() {
+    const unsetIdentifiers = (obj: { id: string; createdAt: Date; updatedAt: Date }) => {
+      delete obj.id
+      delete obj.createdAt
+      delete obj.updatedAt
+    }
+
+    unsetIdentifiers(this.application.accessibility)
+    unsetIdentifiers(this.application.applicant)
+    unsetIdentifiers(this.application.mailingAddress)
+
+    if (this.application.alternateAddress) unsetIdentifiers(this.application.alternateAddress)
```

Comment:
Are sure that it works? I think `alternateAddress` is loaded from the blackApplication.ts and it's an object, so it always has values (?) NOT SURE - just to check :) 

The similar situation below with `alternateContact `

Reply:
Everything's coming directly from the backend in an autofill scenario (which is only when this code gets run), so I'm trying to strip out or change data to fit that flow.